pkg/driver/aws-efs/aws_efs.go

```diff
@@ -102,11 +103,16 @@ func GetAWSEFSOperatorConfig() *config.OperatorConfig {
 // after a client connection + cluster flavour are established.
 func GetAWSEFSOperatorControllerConfig(ctx context.Context, flavour generator.ClusterFlavour, c *clients.Clients) (*config.OperatorControllerConfig, error) {
 	cfg := operator.NewDefaultOperatorControllerConfig(flavour, c, "AWSEFS")
-	cfg.AddDeploymentHookBuilders(c, withCABundleDeploymentHook, withFIPSDeploymentHook)
+	cfg.AddDeploymentHookBuilders(c, withCABundleDeploymentHook, withFIPSDeploymentHook, withCustomTags)
```

Comment:
```suggestion
	cfg.AddDeploymentHookBuilders(c, withCABundleDeploymentHook, withFIPSDeploymentHook)
	if flavour == generator.FlavourHyperShift {
		// hcp-only hooks
		cfg.AddDeploymentHookBuilders(c, withCustomTags)
	}
```
If the `withCustomTags` is used for hcp only, better to only add the hook for hcp.
cc to @dobsonj @gnufied

Reply:
Hey @Phaow `flavour == generator.FlavourHyperShift` would not work for the efs operator as there is only a single manifest( [Slack Thread for reference](https://redhat-internal.slack.com/archives/C057L1GKYSK/p1738941712139239?thread_ts=1738081635.004619&cid=C057L1GKYSK) ). Also the EBS operator also [supports this](https://github.com/openshift/csi-operator/blob/main/pkg/driver/aws-ebs/aws_ebs.go#L378) .

Reply:
For efs operator we check the operator is a hyprshift cluster or not based on the `hypershift.openshift.io/managed: "true"` label and it seems like a bad idea to make requests to the API server during operator initialization. Instead, we always let the controller start, and then Sync in aws_efs_tags_controller.go can return if the infastructure does not have that label.

Reply:
Good to know, thanks for the clarify!